ethereum/core/src/main/java/tech/pegasys/teku/core/epoch/EpochProcessorUtil.java

```diff
@@ -318,28 +314,26 @@ public static void process_registry_updates(MutableBeaconState state)
    *     <a>https://github.com/ethereum/eth2.0-specs/blob/v0.8.0/specs/core/0_beacon-chain.md#slashings</a>
    */
   public static void process_slashings(MutableBeaconState state) {
-    UnsignedLong epoch = get_current_epoch(state);
-    UnsignedLong total_balance = get_total_active_balance(state);
+    UInt64 epoch = get_current_epoch(state);
+    UInt64 total_balance = get_total_active_balance(state);
 
     SSZList<Validator> validators = state.getValidators();
     for (int index = 0; index < validators.size(); index++) {
       Validator validator = validators.get(index);
       if (validator.isSlashed()
           && epoch
-              .plus(UnsignedLong.valueOf(EPOCHS_PER_SLASHINGS_VECTOR / 2))
+              .plus(UInt64.valueOf(EPOCHS_PER_SLASHINGS_VECTOR / 2))
               .equals(validator.getWithdrawable_epoch())) {
-        UnsignedLong increment = EFFECTIVE_BALANCE_INCREMENT;
-        UnsignedLong penalty_numerator =
+        UInt64 increment = EFFECTIVE_BALANCE_INCREMENT;
+        UInt64 penalty_numerator =
             validator
                 .getEffective_balance()
                 .dividedBy(increment)
                 .times(
-                    min(
-                        UnsignedLong.valueOf(
-                            state.getSlashings().stream().mapToLong(UnsignedLong::longValue).sum()
-                                * 3),
-                        total_balance));
-        UnsignedLong penalty = penalty_numerator.dividedBy(total_balance).times(increment);
+                    UInt64.valueOf(
+                            state.getSlashings().stream().mapToLong(UInt64::longValue).sum() * 3)
```

Comment:
We should review this to make sure it can never overflow. But that's not for this PR.

Reply:
This was dealt with in the spec in https://github.com/ethereum/eth2.0-specs/pull/1286 and the implementation here looks to match the spec. So I believe we are good.